packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerViewDesktop.tsx

```diff
@@ -180,7 +180,10 @@ export function DateRangePickerViewDesktop<TDate>(props: DesktopDateRangeCalenda
         const monthOnIteration = utils.setMonth(currentMonth, utils.getMonth(currentMonth) + index);
 
         return (
-          <DateRangePickerViewDesktopContainer key={index}>
+          <DateRangePickerViewDesktopContainer
+            key={index}
+            data-mui-test={`date-range-picker-${index}`}
```

Comment:
I'm always suspicious about adding `data-` attributes for testing, because if you can not describe a component with a selector, it may be impossible for the screen reader too 🤔

At least, could ba renamed as follow, because I first thought "what do we have multiple date range picker"
```suggestion
            data-mui-test={`date-range-picker-calendar-${index}`}
```

Reply:
I've added it to have some clear way to target day by the number/label on a specific calendar, because after these changes and the fact, that `columnheader` value does not tie into `getAccessibleName` calculation there is no easy way to specify, which actual date we want to select.  
One other way, which could work is selecting element by `month year` label and getting it's parent - the appropriate calendar we want.